### PR TITLE
fix: Don't show schema for schema-less plugins

### LINKED ISSUE #25413: [Feature]: Don’t show schema section for plugins that don’t support it
### Is there an existing issue for this?

- [X] I have searched the existing issues

source: https://docs.google.com/spreadsheets/d/1QOXj3w4QuFkG9N7fwfYsOgErrXZn5d9VZleO3XId8zQ/edit#gid=0&range=3:3

notion: https://www.notion.so/appsmith/Activation-60c64894f42d4cdcb92220c1dbc73802?p=78f82b6abf7948c5a7d596ae583ed8a4&pm=s

### Summary

for plugins that don’t support getstructure, we should not show the `Schema` section

List of plugins:
@Irongade please add the list here

### Why should this be worked on?

activation project

### PR DESCRIPTION
**Pull request by Irongade**
This PR hides schemas for actions of schema-less plugins in the query editor.

Fixes #25413 

- Bug fix (non-breaking change which fixes an issue)

- [x] Manual
- [ ] Jest
- [x] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [x] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/pages/Editor/Explorer/Datasources/DatasourceStructureContainer.tsx [hunk 1]
```diff
@@ -37,6 +38,18 @@ export enum DatasourceStructureContext {
   API_EDITOR = "api-editor",
 }
 
+// leaving out DynamoDB and Firestore because they have a schema but not templates
+export const SCHEMALESS_PLUGINS: Array<string> = [
+  PluginName.SMTP,
```
Comment (sneha122) [inline app/client/src/pages/Editor/Explorer/Datasources/DatasourceStructureContainer.tsx:43]: @Irongade Google sheets does not have schema either, should we add that as well?

Comment (Irongade) [reply app/client/src/pages/Editor/Explorer/Datasources/DatasourceStructureContainer.tsx:43]: Is google sheets not going be worked on this sprint? @sneha122

**Review by sneha122 — DISMISSED** [review]

**Review by sneha122 — APPROVED** [review]

**Pull request merged**